### ISSUE TITLE
[bug] conan info fails if a package was built with --require-override

### ISSUE THREAD

**Issue by jave27** [conan-io/conan#12589]
When building with the `--require-override` flag, `conan info` fails to get any information about the successfully-created package. I'd like to use this for our internal builds of libraries that are using outdated (and possibly CVE-impacted) versions of `openssl` or `zlib`. Ultimately, the solution is to update all of the recipes on conan-center to use the new versions of their dependencies, but as a stop-gap, I was hoping to use the override mechanism. Not having access to `conan info` on these packages is a hinderance.

### Environment Details (include every applicable attribute)
  * Ubuntu 18.04
  * gcc 7.5.0
  * Conan version: 1.54.0
  * Python version: 3.10.7

### Steps to reproduce (Include if Applicable)
1. Setup a local conan server as your only remote.
2. Use the conan-center-index recipes to build `openssl/1.1.1s`, and publish it to your remote.
3. Build `ldns/1.8.1` (which at the moment requires `openssl/1.1.1o`) with the command: `conan create ./conan-center-index/recipes/ldns/all ldns/1.8.1@ --require-override=openssl/1.1.1s`. The Build should succeed.
4. Now type `conan info ldns/1.8.1@`

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
openssl/1.1.1o: Retrieving from server 'myremote' 
openssl/1.1.1o: Trying with 'myremote'...
ERROR: Failed requirement 'openssl/1.1.1o' from 'ldns/1.8.1'
ERROR: openssl/1.1.1o was not found in remote 'myremote'
```

**Comment by memsharded** [comment]
Hi @jave27 

I have a quick question, why are you running a ``conan info`` just after the ``conan create``? 
Have you considered running the ``conan create --json=myjsonfile``? That will create a json with a lot of information about the dependency graph, probably something close to what ``conan info`` is retrieving.

**Comment by jave27** [comment]
Yes, I could get that information from `create` itself, but that doesn't change the fact that if I wanted that info at some other point in the future, I believe it is inaccessible unless I rebuild the package.

Well, inaccessible programmatically anyway. Looking at the generated `conaninfo.txt` file in the package folder itself, it seems like that information is all there.

**Comment by memsharded** [comment]
I see. It looks like a missing feature, it was never implemented on ``conan info``, because we were never confident in the ``--require-override`` approach. In fact it has been removed in 2.0, and it will only accept overrides either on recipes, or partially in lockfiles (as long as the override fits in a defined version range, lockfiles will not be able to override outside of defined version ranges). The idea is that it is not possible to substantially modify the dependency graph from command line only without having it in code (either recipe, or a lockfile), because we have also learned that overriding the upstream dependencies quickly become a nightmare to developers using them, because things start to fail, and it is complicate to know what version or versions have to be passed on the command line.

At this stage in 2.0 (already preparing beta.6, if things go well, will be the last one), it is unlikely that this feature will be implemented in 1.X, so I'd recommend giving the ``json`` a try. Thanks!
